includes/Geocoders.php

```diff
@@ -153,45 +144,18 @@ public static function geocode( $address, $geoService = '' ) {
 		if ( $geocoder === false ) {
 			return false;
 		}
-
-		return self::getGeocoded( $geocoder, $address );
-	}
-
-	/**
-	 * Returns already coordinates already known from previous geocoding operations,
-	 * or false if there is no match found in the cache.
-	 *
-	 * @since 0.7
-	 *
-	 * @param string $address
-	 *
-	 * @return LatLongValue|boolean false
-	 */
-	private static function cacheRead( $address ) {
-		global $egMapsEnableGeoCache;
-
-		if ( $egMapsEnableGeoCache && array_key_exists( $address, self::$globalGeocoderCache ) ) {
-			return self::$globalGeocoderCache[$address];
-		}
-
-		return false;
-	}
-
-	/**
-	 * Writes the geocoded result to the cache if the cache is on.
-	 *
-	 * @since 0.7
-	 *
-	 * @param string $address
-	 * @param LatLongValue $coordinates
-	 */
-	private static function cacheWrite( $address, LatLongValue $coordinates ) {
-		global $egMapsEnableGeoCache;
-
-		// Add the obtained coordinates to the cache when there is a result and the cache is enabled.
-		if ( $egMapsEnableGeoCache && $coordinates ) {
-			self::$globalGeocoderCache[$address] = $coordinates;
-		}
+		
+		$caching = new Geocoders\CachingGeocoder( 
+			$geocoder, 
+			wfGetCache( CACHE_ANYTHING )
+		);
```

Comment:
Like this for every call to the geocode function, a new instance of the CachingGeocoder is created. Perhaps not a huge deal, but there is some irony here :)

Reply:
Aha indeed, that's funny ^^ But doing it in another way would complicate things, no ? I don't see a nice way actually, as the input `Geocoder` might change at each call...